### PR TITLE
Validation feature

### PR DESCRIPTION
**Pull request by YannickZahinda**
### Blog_App : Validation

In this project I performed validation formy project : 
- For the user model
- For the Post model

### CODE REVIEW

**Review by peterrobert — CHANGES_REQUESTED** [review]
Hi @YannickZahinda, 

Very nice project. Good work making validations on the models👏🏿 . However, there are a few changes to be made kindly check below ⬇️

### Highlights!

[ ] Good feature branch name ✔️
[ ] Descriptive pull request ✔️

 ## Required Changes ♻️
_Check the comments under the review._

- [ ] It will be great to write the unit tests for `User` , `Post`, `Comment` and `Like` this is a major requirement for this project. It seems you haven't written the tests yet. Example: 
```ruby
describe 'validations' do
  it { should validate_presence_of(:name) }
end
```
Kindly make this change and submit you project for another review.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- It is always good to capitalize the first letter of your commit messages. This makes your project even more professional.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or tag me @peterrobert if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  it 'CommentsCounter should not be below 0' do
+    subject.comment_counter = -1
+    expect(subject).to_not be_valid
+  end
+
+  it 'LikesCounter should not be below 0' do
+    subject.likes_counter = -1
+    expect(subject).to_not be_valid
+  end
```
Comment (Deepakdanger) [inline spec/models/post_spec.rb:42]: - [ ] Please add the positive case as well where the posts are valid.

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  end
+
+  it 'PostsCounter should not be below 0' do
+    subject.posts_counter = -1
+    expect(subject).to_not be_valid
+  end
+
+  it 'PostsCounter should not be a string' do
+    subject.posts_counter = 'My string'
+    expect(subject).to_not be_valid
```
Comment (Deepakdanger) [inline spec/models/user_spec.rb]: - [ ] Please add the positive case as well where the users are valid.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @YannickZahinda, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

- [ ] As requested by the previous reviewer, Please add the unit tests for `Comment` and `Like`. Also, add both positive and negative scenarios.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by andersonlebon — APPROVED** [review]
## Hi @YannickZahinda 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 

## Optional suggestions
- [ ] Kindly Check the comments under the review._
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**